### PR TITLE
Support autoredirection

### PR DESCRIPTION
**Pull request by Jeff-SearchPilot**
This allows the user to pass in the arn as a url argument and be immediately redirected to it.

### CODE REVIEW

**Comment by fxkr** [comment]
Thank you for the contribution. This is an interesting idea.

I apologize in advance that this will take me some time to review. I will need to carefully think about what input validation link2aws is doing and needs to do before we merge this, as this can have security implications (Open Redirect Abuse: [link](https://www.fastly.com/blog/open-redirects-real-world-abuse-and-recommendations)/[link](https://cheatsheetseries.owasp.org/cheatsheets/Unvalidated_Redirects_and_Forwards_Cheat_Sheet.html)).

A few more review thoughts:

* Privacy: this uses query parameters (`?arn=...`), which will leak to the server (GitHub, so maybe not really an actual issue). For privacy reasons, it might be nice to use fragments instead (`#...`), which aren't sent to the server. But we can support both, and for this PR, `?arn=` would be fine.

* Browser compatibility: I wasn't sure, but `URLSearchParams` seems to be [widely supported](https://caniuse.com/?search=urlsearchparams). So no problem.

* Also relevant: #8

**Comment by Jeff-SearchPilot** [comment]
Any further thoughts on this @fxkr?

**Comment by fxkr** [comment]
Hi @Jeff-SearchPilot , sorry for the delay. I have been traveling and haven't had any time to look at this yet.

File: index.html [hunk 1]
```diff
@@ -101,4 +113,4 @@ <h1>link2aws.github.io</h1>
     </script>
 </body>
 
-</html>
```
Comment (fxkr) [inline index.html]: Can you remove the unnecessary whitespace change?

File: index.html [hunk 2]
```diff
@@ -63,6 +63,18 @@ <h1>link2aws.github.io</h1>
 
             //document.getElementById("text").value = "arn:aws:iam::aws:policy/AdministratorAccess";
             onUpdate();
+
+            // If the user passes in the arn as a parameter, immediately redirect them
+            var params = new URLSearchParams(window.location.search);
```
Comment (fxkr) [inline index.html]: Would you be ok with switching to window.location.hash?

I am asking for this change because this way, the ARN will not be sent to the web server. This is better for privacy.

Please let me know if you see any reason against window.location.hash. If there are any technical problems with it, then we can keep window.location.search. But if they are equal then I would prefer the more privacy friendly option.

**Review by fxkr — CHANGES_REQUESTED** [review]
Hi @Jeff-SearchPilot , I have now merged a commit with extra input validation, so that we can go ahead with this now. Can you address the following review comments?

Thank you very much!

**Review by fxkr — APPROVED** [review]

**Comment by fxkr** [comment]
Thanks! Merged.
